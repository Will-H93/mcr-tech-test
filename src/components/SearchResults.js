import PropTypes from "prop-types";
import React from "react";
import "../styles/search-results.css";

const SearchResults = ({ results }) => {
  if (!results.length) {
    return <p>No results</p>;
  } else {
    return (
      <div className="search-results">
        {results.map((image) => (
          <img
            className="card-image"
            src={image}
            alt="spaceImage"
            key={image}
          />
        ))}
      </div>
    );
  }
};

SearchResults.propTypes = {
  results: PropTypes.arrayOf(PropTypes.string).isRequired,
};

export default SearchResults;
