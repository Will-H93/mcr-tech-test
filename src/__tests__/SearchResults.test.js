import React from "react";
import { render, screen } from "@testing-library/react";
import SearchResults from "../components/SearchResults";

describe("SearchResults", () => {
  const validProps = {
    results: ["test-url-1", "test-url-2", "test-url-3"],
  };

  test("renders correctly", () => {
    const { asFragment } = render(<SearchResults {...validProps} />);

    expect(asFragment()).toMatchSnapshot();
  });

  test("renders an image", () => {
    render(<SearchResults {...validProps} />);
    const image = screen.getAllByRole("img");

    image.forEach((image) => {
      expect(image).toHaveAttribute("src");
      expect(image.src !== "").toBeTruthy();
    });
  });
});
