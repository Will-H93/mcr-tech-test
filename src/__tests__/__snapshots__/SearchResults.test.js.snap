// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults renders correctly 1`] = `
<DocumentFragment>
  <div
    class="search-results"
  >
    <img
      alt="spaceImage"
      class="card-image"
      src="test-url-1"
    />
    <img
      alt="spaceImage"
      class="card-image"
      src="test-url-2"
    />
    <img
      alt="spaceImage"
      class="card-image"
      src="test-url-3"
    />
  </div>
</DocumentFragment>
`;
